src/qml/editorwidgets/RelationEditor.qml

```diff
@@ -0,0 +1,239 @@
+                      anchors.margins: 4 * px
+                      fillMode: Image.PreserveAspectFit
+                      horizontalAlignment: Image.AlignHCenter
+                      verticalAlignment: Image.AlignVCenter
+                      source: Style.getThemeIcon( 'ic_delete_forever_white_24dp' )
+                    }
+                }
+
+                onClicked: {
+                    deleteDialog.referencingFeatureId = model.referencingFeature.id
+                    deleteDialog.visible = true
+                }
+            }
+          }
+
+          //bottom line
+          Rectangle {
+            id: bottomLine
+            anchors.bottom: parent.bottom
+            height: 1
+            color: "lightGray"
+            width: parent.width
+          }
+        }
+    }
+
+    //the delete entry stuff
+    MessageDialog {
+      id: deleteDialog
+
+      property int referencingFeatureId
+      property var layerName
+
+      visible: false
+
+      title: "Delete feature "+referencingFeatureId+" on layer "+layerName //translation needed
+      text: "Should the feature "+referencingFeatureId+" on layer "+layerName+" really be deleted?" //translation needed
+      standardButtons: StandardButton.Ok | StandardButton.Cancel
+      onAccepted: {
+        referencingFeatureListView.model.deleteFeature( referencingFeatureId )
+        console.log("delete feature "+referencingFeatureId)
+        visible = false
+      }
+      onRejected: {
+        visible = false
+      }
+    }
+
+    //the add entry stuff
+    Loader {
+      id: embeddedFeatureForm
+
+      property var state
+      property var relatedFeature
+      property var relatedLayer
+
+      sourceComponent: embeddedFeatureFormComponent
+      active: false
+      onLoaded: {
+        item.open()
+      }
+    }
+
+    Component {
+      id: embeddedFeatureFormComponent
+
+      Popup {
+        id: popup
+        parent: ApplicationWindow.overlay
+
+        x: 24 * dp
+        y: 24 * dp
+        width: parent.width - 48 * dp
+        height: parent.width - 48 * dp
+        modal: true
+        focus: true
+        closePolicy: Popup.CloseOnEscape
+
+        FeatureForm {
+            model: AttributeFormModel {
+            id: attributeFormModel
+
+            featureModel: FeatureModel {
+              currentLayer: relatedLayer
+              feature: state != "Add" ? embeddedFeatureForm.relatedFeature : undefined
+              linkedParentFeature: relationEditorModel.feature
+              linkedRelation: relationEditorModel.relation
+            }
+          }
+
+          embedded: true
+          toolbarVisible: true
+
+          anchors.fill: parent
+
+          state: embeddedFeatureForm.state
+
+          onSaved: {
+            popup.close()
+          }
+
+          onCancelled: {
+            popup.close()
+          }
+        }
+
+        onClosed: {
+          embeddedFeatureForm.active = false
+          relationEditorModel.reload()
```

Comment:
What's this for?

Reply:
When a feature is deleted, it should reload the model - but you are right it's doing it anyway, so I can remove it.

Reply:
Oh it's on adding a feature. Then it needs to be reloaded and I cannot remove it.